sys/net/gnrc/transport_layer/tcp/gnrc_tcp_fsm.c

```diff
@@ -33,6 +33,11 @@
 #include "debug.h"
 
 /**
+ * @brief Flag indicating if a thread blocked by the GNRC TCP Api must be notified.
+ */
+static bool _notify_user;
```

Comment:
I think you could have a specific return value/state for this, so instead of an error code you could have 

```
#define GNRC_TCP_FSM_NOTIFY_USER    (42U)
```